tests/test_experiment.py

```diff
@@ -389,3 +393,135 @@ def impl(*a, **kw):
     assert len(calls) == 1, "Launcher's `get_status` was called more than once"
     (call,) = calls
     assert call == ((id_,), {}), "IDs were not de-duplicated"
+
+
+def test_wait_handles_empty_call_args(experiment):
+    """An exception is raised when asked to wait for the completion of no jobs"""
+    with pytest.raises(ValueError, match="No job ids"):
+        experiment.wait()
+
+
+def test_wait_does_not_block_unknown_id(experiment):
+    """If an experiment does not recognize a job id, it should not wait for its
+    completion
+    """
+    now = time.perf_counter()
+    experiment.wait(dispatch.create_job_id())
+    assert time.perf_counter() - now < 1
+
+
+def test_wait_calls_prefered_impl(make_populated_experment, monkeypatch):
+    """Make wait is calling the expected method for checking job statuses.
+    Right now we only have the "polling" impl, but in future this might change
+    to an event based system.
+    """
+    exp = make_populated_experment(1)
```

Comment:
typo: change to "make_populated_experiment"

Reply:
I promise I will learn english one day!